cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,220 @@
+
+  // Create thread pool
+  private static ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+  private static class SingletonTracer {
+
+    private static final Tracer TRACER = InitTracer();
+
+    private SingletonTracer() {
+    }
+
+    private static Tracer InitTracer() {
+      Tracer tracer = Tracing.getTracer();
+
+      try {
+        // Set up StackDriver
+        StackdriverTraceExporter.createAndRegister(
+            StackdriverTraceConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+      } catch (java.io.IOException e) {
+        LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+      }
+
+      // Trace every request
+      TraceConfig traceConfig = Tracing.getTraceConfig();
+      traceConfig.updateActiveTraceParams(
+          traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample())
+              .build());
+
+      return tracer;
+    }
+
+
+    private static Tracer getTracer() {
+      return SingletonTracer.TRACER;
+    }
+  }
+
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+    String connType = request.getFirstQueryParameter("conn_type").orElse("pool");
+
+    String requestId = UUID.randomUUID().toString();
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    Runnable connectFunc;
+    switch (connType) {
+      case "pool":
+        connectFunc = () -> connectWithPool(createPool(jdbcURL, connProps), requestId);
+        break;
+      case "regular":
+        connectFunc = () -> connectRegular(jdbcURL, connProps, requestId);
+        break;
+      default:
+        response.setStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+        BufferedWriter writer = response.getWriter();
+        writer.write("Valid options for conn_type are: pool, regular");
+        return;
+    }
+
+    ScheduledFuture<?> task = execService
+        .scheduleAtFixedRate(connectFunc, 0, interval, TimeUnit.MILLISECONDS);
+
+    try {
+      TimeUnit.MILLISECONDS.sleep(duration);
+    } catch (InterruptedException e) {
+      response.setStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+      BufferedWriter writer = response.getWriter();
+      writer.write(ExceptionUtils.getStackTrace(e));
+      return;
+    }
+
+    task.cancel(true);
```

Comment:
you probably want to do this even if InteruptedException occurs?

Reply:
fixed!